### ISSUE TITLE
Ai behaviour

### ISSUE THREAD

**Issue by RichCoop** [kennyhx2/Submarine-Titans-AI#2]
Greetings, recently i ve managed to give this mod a try, it made the AI smarter indeed, but not without flaws.
Now I will tell my expeerience with Death Match, FFA. But i believe it suits TDM aswell.
First i'd like to tell about this teleportation feature in this mod. This is ridiculois, if you play 1 vs 1 it's OK, but if there are more it is just insane, AI starts to teleport troops (usually with marauders or something) to capture your buildings, maybe I am that lucky, but it begins before i get any tools to prevent the teleports. And since AI forms alliances, i get teleport spam all the time. Can there also be a feature to disable this teleport for AI? Like another difficulty or something? This teleportation is really annoying.
Another issue with this is the fleet behaviour. AI doesn't make somehow strong fleets and does not attack with them. Yes, it has some subs, enough to strike down wild scouts, but not enough to break player's base. It doesn't even try, it just repeats the teleportation of ~4 subs again and again. So the gameplay looks this way (in FFA), you hold the line on your base, try to survive teleport spam, then once AI will have low resources the spam becomes less frequent and you either wait untill AI have no resources at all or just wait untill it have low of them then you try to penetrate the defence. 
Oh yes, AI does build it, and it abuses the alliance, when you try to invade one AI player, you see the defensive buildings of every other AI player aswell. And as good as it sounds it's also a problem. AI invests too much into the defence. So in the late game AI just sits and waits for you. So you have to slowly push forward. That's not much entertaining. So again, it would be very nice for this mod to also have a difficulty level where AI does not have the mass teleportation feature, and maybe invest more into its fleets.

P.S. Beat 4 player FFA as WS by exhausting AI.

**Comment by Kander2616** [comment]
+1 
I also want to add why do you think the technology to repel weapons of mass destruction is flawed? I think it was created to serve certain percentages and not 100%

I think that if something is in the original game, do not remove it since this is a mod for artificial intelligence, not to eliminate structures or submarines

I really liked how artificial intelligence teleports but it is exaggerated and unrealistic since they should use the structure to teleport

translated with google translator

**Comment by RichCoop** [comment]
Another several things spotted during the gameplay.
1. Silicons AI cheats. They receive energy and silicon from nowhere whenever it hits zero. BO and WS don't do that.
2. AI wastes too much resources into the defence and relies too much on the teleportation. I played as silicon with WS ally and fed him with resources, he grew somehow solid army, but still couldnt beat anyone. Opponents got too big defence line for such small fleet amount AI teleports for the attack. AI should spend more stuff into the army at least when the defence is somehow good, and then move the forces to the attack. What i see now is if AI gets resources from me, he just constantly aims to build another HF cannon, another psycothron, but no more subs. Also WS seems to love BO subs too much. Even with full upgrades WS got heavy cruisers and destroyers (!) but no cruisers of its own. Maybe some limitations for the stolen subs is required?
3. Weapons of mass destruction is prohibited? Then why do Silicons got their Gas Satellite???? Its not as powerful as the Vaccuum Bomb, sure, but still it is a remote thing enemy has no defence from. I'd return the mass weapons, but prohibit it for AI only if its such a big issue.
4. Silicons and their recyclotron, it is not a mod thing, but in the game itself it is sooooo broken. If this mod restricts things, it should also restrict recyclotron or nerf it and resource exchange of Silicons.
5. AI really loves to build silos, even if he doesn't have mines already. For 4 mines ai can easily build up to 8 silos. Why? If you destroy the silo or capture it, AI will cosntruct another one and again and again untill it hits the number ai is fine with. But in the same time if you destroy the market, ai won't be so fast re rebuild it or wont rebuild it at all.
6.  BO and WS seem to never build a mine launcher and plasma cannon, why?

**Comment by RichCoop** [comment]
Tested it again, found some other interesting issues.
1. BO and WS starting build order. They build Air/O2 too late. First they plae 2 docks/subcenters, then labs and only then they place air. Mostly they start to build it near 2nd minute of the game. This is somehow slows them down and gives a big advantage to silicons. This also makes them very vulnerable for early rush in 1vs1 games.
2. Silicons AI is too overpowered, it wrecks WS and BO AIs too easily. As i mentioned in my previous message they cheat with resources, what, i believe, is the reason of that power.
3. Humans build 5 labs, i think this is too much. 3 or 4 would be just enough. A 5th lab almost always does nothing, but AI spends resources to build it, and if you break the lab the AI will continiously try to rebuild it. Is there a way, to stop labs production once AI learns a certain tech?
4. Again, as mentioned before, WS AI likes BO subs and buildings too much, once it gets the BO's tech, WS become BO at that point. 
5. Is there a way to restrict WS of building the Phantom revealing buildings and subs, when there are no BO players in the game? As it is a shame WS waste resources of them, when there re no invisible units?

**Comment by knuxyl** [comment]
I agree with just about everything above. And to add, on a flagship run with a teammate ai, the teammate never attacks the enemy, it just goes around finding resources. when it's out of resources, it just patrols the area it has structures.

**Comment by dinoomanovic** [comment]
Thanks for the mod. One thing I noticed is that in Battles there is no option for AI assistant, it says None. Can it be added back?
